src/App.tsx

```diff
@@ -15,36 +16,114 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_ALPHABETICALLY = 'alphabet',
+  SORT_BY_LENGTH = 'length',
+  SORT_REVERSE = 'reverse',
+  SORT_RESET = '',
+}
+type Goods = string[];
+
+function getPreparedGoods(
+  goods: Goods,
+  sortField: SortType,
+  isReverse: SortType,
+): Goods {
+  let preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.SORT_BY_ALPHABETICALLY:
+          return good1.localeCompare(good2);
+        case SortType.SORT_BY_LENGTH:
+          return good1.length - good2.length;
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    preparedGoods = preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState(SortType.SORT_RESET);
+  const [isReverse, setIsReverse] = useState(SortType.SORT_RESET);
```

Comment:
```suggestion
  const [isReverse, setIsReverse] = useState(false);
```
you can just save the boolean value here